### ISSUE TITLE
[arm32/Ubuntu] Assert failure: Consistency check failed: hit privileged instruction!

### ISSUE THREAD

**Issue by BruceForstall** [dotnet/coreclr#17610]
In GCStress=0xc

https://ci.dot.net/job/dotnet_coreclr/job/master/job/jitstress/job/arm_cross_checked_ubuntu_gcstress0xc_tst/2/consoleText

```
FAILED   - baseservices/threading/generics/WaitCallback/thread23/thread23.sh
               BEGIN EXECUTION
               /ssd/j/workspace/dotnet_coreclr/master/jitstress/arm_cross_checked_ubuntu_gcstress0xc_tst/bin/tests/Linux.arm.Checked/Tests/Core_Root/corerun thread23.exe
               
               Assert failure(PID 13815 [0x000035f7], Thread: 13888 [0x3640]): Consistency check failed: hit privileged instruction!FAILED: !ExecutionManager::IsManagedCode(GetIP(pContext))
                   File: /mnt/j/workspace/dotnet_coreclr/master/arm_cross_checked_ubuntu/src/vm/excep.cpp Line: 6856
                   Image: /ssd/j/workspace/dotnet_coreclr/master/jitstress/arm_cross_checked_ubuntu_gcstress0xc_tst/bin/tests/Linux.arm.Checked/Tests/Core_Root/corerun
               
               ./thread23.sh: line 244: 13815 Aborted                 $_DebuggerFullPath "$CORE_ROOT/corerun" $ExePath $CLRTestExecutionArguments
               Expected: 100
               Actual: 134
               END EXECUTION - FAILED
```

Related to https://github.com/dotnet/coreclr/issues/17585 and/or https://github.com/dotnet/coreclr/issues/17598?

**Comment by sdmaclea** [comment]
Agreed

**Comment by AndyAyersMS** [comment]
This is one of the gc stress races I am trying to fix with #17330. The race is not arch or platform specific.

**Comment by AndyAyersMS** [comment]
Should be fixed via #17330.
